pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +272,27 @@ func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatu
 			},
 		}
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodHostIPs) {
```

Comment:
we don't check gates in conversion

Reply:
This is still necessary. After the feature is turned off, the hostIPs is nil. If the gate is not added, it will be filled in here.

Reply:
https://github.com/kubernetes/kubernetes/pull/101566/commits/05273bfc03cf60780728381b3729d695070b5d60
https://github.com/kubernetes/kubernetes/pull/101566/commits/a22f0c71f9c6e2fbbf9af597cf3ab759fdf9c99e

These two commits, enable the feature to fill in for some without HostIPs, or disable the feature to drop for some with HostIPs

Reply:
> After the feature is turned off, the hostIPs is nil. If the gates is not added, it will be filled in here.

if hostsIPs is nil then len(in.HostIPs) == 0, do we really need the feature gate?